### ISSUE TITLE
Integrity without authentication?

### ISSUE THREAD

**Issue by ekr** [stackevo/spudreq#6]
"Integrity: : SPUD must provide integrity to detect modifications of information that are not supposed to be changed deliberately or non-deliberately by (SPUD-aware or not-SPUD-aware) middleboxes."

This comes right after saying that authentication is not required. I think you must mean to
say that authentication of middleboxes is not required, since integrity basically requires
data origin authentication.

**Comment by mirjak** [comment]
Yes, the main point about authentication was that no authentication of middleboxes should be required. 

However, this point about integrity is probably a little too short. One point is that one should be able to detect if information was erroneously changed by a non-SPUD aware middlebox; here a checksum is probably enough. The other point is intentionally modifying provided information. Here authentication is needed. And in this case there is again a difference between authenticating one endhost or any middlebox on the path. And there might be cases where it is actually intended that one middlebox can overwrite information of another middlebox. That's definitely something to discuss.
